examples/double-proxy/envoy-backend.yaml

```diff
@@ -0,0 +1,54 @@
+static_resources:
+  listeners:
+  - name: postgres_listener
+    address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 5432
+    listener_filters:
+    - name: "envoy.filters.listener.tls_inspector"
+      typed_config: {}
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.postgres_proxy
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.postgres_proxy.v3alpha.PostgresProxy
+          stat_prefix: egress_postgres
+      - name: envoy.filters.network.tcp_proxy
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+          stat_prefix: postgres_tcp
+          cluster: postgres_cluster
+      filter_chain_match:
+        server_names:
+          - proxy-postgres-backend.example.com
```

Comment:
This is actually an example of SNI routing as I mentioned in my other PR comments. Maybe explicitly call it out as such?

Reply:
ah great - i think im starting to understand the relationship between mtls, sni and dns

im going to add the page i mentioned elsewhere to quick-start about "securing envoy" so i can straighten out in my mind each of the components. 

hopefully this can still land quite quickly, but ill use the confs here to test what im writing etc...

Reply:
@mattklein123 reading through this sandbox again, my thought is actually to remove SNI from this example.

my reason for this is to keep it simple. It already has quite a lot of cert/key complexity, and now  that sni is documented elsewhere, here doesnt seem the best place for it.

lmk your thoughts, and i can update either way

Reply:
also - see https://github.com/envoyproxy/envoy/pull/13844#issuecomment-724557562